pkg/util/resize/configspec.go

```diff
@@ -269,6 +270,40 @@ func compareExtraConfig(
 	outCS.ExtraConfig = util.MergeExtraConfig(ci.ExtraConfig, extraConfig)
 }
 
+// compareConsolePreferences compares the console preferences settings in the Config Spec.
+func compareConsolePreferences(
+	ci vimtypes.VirtualMachineConfigInfo,
+	cs vimtypes.VirtualMachineConfigSpec,
+	outCS *vimtypes.VirtualMachineConfigSpec) {
+
+	if cs.ConsolePreferences == nil {
+		return
+	}
+
+	if ci.ConsolePreferences == nil {
+		// If configInfo console preferences is nil, there is nothing to compare.
+		// set desired values
+		outCS.ConsolePreferences = &vimtypes.VirtualMachineConsolePreferences{
+			PowerOnWhenOpened:        cs.ConsolePreferences.PowerOnWhenOpened,
```

Comment:
Shouldn't this also check that all field here are not-nil, right? Ideally we don't want just `outCS.ConsolePreferences  = &vimtypes.VirtualMachineConsolePreferences{}` because that is going to cause an unnecessary Reconfigure.

Reply:
yep.. that would be better. updated the comment to reflect the same logic